ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -320,56 +340,118 @@ def handleInput(self, paramInput):
+    
+    ####################################################################################
+    # reproduction node                                                                #
+    ####################################################################################
     reproductionNode = gaParamsNode.findFirst('reproduction')
     self._nParents = int(np.ceil(1/2 + np.sqrt(1+4*self._populationSize)/2))
     self._nChildren = int(2*comb(self._nParents,2))
-    # crossover node
+    
+    ####################################################################################
+    # crossover node                                                                   #
+    ####################################################################################
     crossoverNode = reproductionNode.findFirst('crossover')
     self._crossoverType = crossoverNode.parameterValues['type']
+    if self._crossoverType not in ['onePointCrossover','twoPointsCrossover','uniformCrossover']:
+      self.raiseAnError(IOError, f'Currently constrained Genetic Algorithms only support onePointCrossover, twoPointsCrossover and uniformCrossover as a crossover, whereas provided crossover is {self._crossoverType}')
     if crossoverNode.findFirst('points') is None:
       self._crossoverPoints = None
     else:
       self._crossoverPoints = crossoverNode.findFirst('points').value
     self._crossoverProb = crossoverNode.findFirst('crossoverProb').value
     self._crossoverInstance = crossoversReturnInstance(self,name = self._crossoverType)
-    # mutation node
+    
+    ####################################################################################
+    # mutation node                                                                    #
+    ####################################################################################
     mutationNode = reproductionNode.findFirst('mutation')
     self._mutationType = mutationNode.parameterValues['type']
+    if self._mutationType not in ['swapMutator','scrambleMutator','inversionMutator','bitFlipMutator','randomMutator']:
+      self.raiseAnError(IOError, f'Currently constrained Genetic Algorithms only support swapMutator, scrambleMutator, inversionMutator, bitFlipMutator, and randomMutator as a mutator, whereas provided mutator is {self._mutationType}')
     if mutationNode.findFirst('locs') is None:
       self._mutationLocs = None
     else:
       self._mutationLocs = mutationNode.findFirst('locs').value
     self._mutationProb = mutationNode.findFirst('mutationProb').value
     self._mutationInstance = mutatorsReturnInstance(self,name = self._mutationType)
-    # Survivor selection
+    
+    ####################################################################################
+    # survivor selection node                                                          #
+    ####################################################################################
```

Comment:
This is not XML, I'd either remove the word 'node' from the comment or make it '# Reading from survivorSelection node'

Reply:
"node" is now removed.